osbuild/util/rhsm.py

```diff
@@ -5,18 +5,59 @@
 """
 
 import configparser
+import glob
+import os
 import re
 
 
 class Subscriptions:
     def __init__(self, repositories):
         self.repositories = repositories
+        self.fallback_method_available = False
+        self.fallback_secrets = None
```

Comment:
I don't think you need `fallback_method_available`. If `fallback_secrets` has entries, you can just use its truthiness.

Reply:
I think it could also just be called `secrets` and you have a comment for the subscription class saying that this is used, if available, if no match could be found via `repositories`.

Reply:
When I was reading the code again, I realized this could be just left out so sure, I'll get rid of it. Now regarding the name, I'd keep the "fallback" in there because it is something we'd like to get rid of. I dislike the way it works, because it is just guessing.

Reply:
Make sense, although I think this would be better suited for a comment than in the variable name.

Reply:
hmm, let's just have it without the fallback in the name